core/src/transaction.rs

```diff
@@ -168,12 +159,12 @@ impl Deref for UnverifiedTransaction {
 }
 
 impl TryFrom<UnverifiedTransaction> for VerifiedTransaction {
-    type Error = &'static str;
-    fn try_from(tx: UnverifiedTransaction) -> Result<Self, Self::Error> {
+    type Error = Error;
+    fn try_from(tx: UnverifiedTransaction) -> Result<Self, Error> {
         if tx.verify_transaction() {
             Ok(VerifiedTransaction(tx.0))
         } else {
-            Err("")
+            Err(Error::Transaction(TransactionError::InvalidVerification))
```

Comment:
I think `InvalidVerification` implies the "verification" itself is invalid. What about "InvalidSignature"?

Reply:
Good idea! I modified it!